osu.Game/Screens/Edit/Screens/Setup/Components/OsuColourPickerGradient.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Configuration;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input;
+using System;
+
+namespace osu.Game.Screens.Edit.Screens.Setup.Components
+{
+    public class OsuColourPickerGradient : Container, IHasCurrentValue<Color4>
+    {
+        private bool isColourChangedFromGradient;
+        private bool isColourChangedFromActiveColour;
+
+        private readonly Container pickerContainer;
+
+        public const float SIZE_X = 180;
+        public const float SIZE_Y = 180;
+        public const float COLOUR_LABEL_TEXT_SIZE = 18;
+
+        public event Action<Color4> SelectedColourChanged;
+
+        public void TriggerSelectedColourChanged(Color4 newValue)
+        {
+            SelectedColourChanged?.Invoke(newValue);
+        }
+
+        private Color4 activeColour;
+        public Color4 ActiveColour
+        {
+            get => activeColour;
+            set
+            {
+                if (value == activeColour)
+                    return;
+
+                activeColour = value;
+                pickerContainer.Clear();
+                loadPicker();
+                isColourChangedFromActiveColour = true;
+                Current.Value = calculateColour(selectedColourPosition);
+                isColourChangedFromActiveColour = false;
+            }
+        }
+
+        public float ActiveHue => Color4.ToHsv(activeColour).X;
+
+        public OsuColourPickerGradient()
+        {
+            CircularContainer colourPickerButton;
+
+            Size = new Vector2(SIZE_X, SIZE_Y);
+
+            Children = new Drawable[]
+            {
+                pickerContainer = new Container
+                {
+                    Size = new Vector2(SIZE_X, SIZE_Y),
+                    CornerRadius = 10,
+                    Masking = true,
+                },
+                colourPickerButton = new CircularContainer
+                {
+                    AlwaysPresent = true,
+                    RelativeSizeAxes = Axes.None,
+                    RelativePositionAxes = Axes.None,
+                    Origin = Anchor.Centre,
+                    Anchor = Anchor.TopLeft,
+                    Size = new Vector2(15),
+                    //Position = new Vector2(50),
```

Comment:
Commented code?

Reply:
Still unfinished as the button doesn't appear anyway. Once done, commented code will have been removed entirely.